firefox-ios/Client/TabManagement/Tab.swift

```diff
@@ -705,7 +710,8 @@ class Tab: NSObject, ThemeApplicable, FeatureFlaggable, ShareTab {
             }
         }
 
-        if let webView, webView.url != nil {
+        // Do not reload from origin for homepage internal URLs
+        if let webView, webView.url != nil && !(InternalURL(webView.url)?.isAboutHomeURL ?? true) {
```

Comment:
```suggestion
        if let webView, webView.url != nil && !(InternalURL(webView.url)?.isAboutHomeURL ?? false) {
```
I think we would probably want to err on the side of assuming this isn't an internal URL so we'd still kick off the reload...? What do you think? 🤔 

Separate note, I don't think we need this extra code since the above chunk of code correctly appended the UUID to the internal home URL in my testing!

Reply:
Awesome thank you for testing this out

Reply:
@ih-codes @lmarceau I tested this branch just as a sanity-check and couldn't replicate the audio/video leak with these changes in place. Just gave this a quick look so far but LMK if you need any additional eyes on this